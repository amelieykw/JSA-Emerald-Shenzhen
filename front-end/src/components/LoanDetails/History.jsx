import React from 'react'
import { View, ScrollView, FlatList, Text } from 'react-native'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import HistoryItem from './HistoryItem'
import style from './HistoryStyle'

const History = ({ loanHistory }) => {
  return (
    <View style={style.body}>
      {loanHistory ? (
        <>
          <Text style={style.title}>History</Text>
          <ScrollView style={{ height: 350 }}>
            <FlatList
              data={loanHistory}
              renderItem={({ item }) => <HistoryItem historyItem={item} />}
              keyExtractor={(_, index) => index.toString()}
            />
          </ScrollView>
        </>
      ) : (
        <View style={style.warning}>
          <Text style={style.warningText}>No available records now</Text>
        </View>
      )}
    </View>
  )
}

History.propTypes = {
  loanHistory: PropTypes.arrayOf(PropTypes.any)
}

History.defaultProps = {
  loanHistory: undefined
}

const mapStateToProps = state => {
  return {
    loanHistory: state.loanList.find(loanItem => loanItem.id === state.detailLoanId).history
  }
}

export default connect(mapStateToProps, null)(History)
