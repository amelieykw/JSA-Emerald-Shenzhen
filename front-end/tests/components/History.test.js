import React from 'react'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import data from '../../helpers/mockData_FE'
import History from '../../src/components/LoanDetails/History'

const mockStore = configureStore([thunk])

describe('<History />', () => {
  it('<History /> should render correctly', () => {
    const store = mockStore({ loanList: data.detailedLoans, detailLoanId: 1 })
    const component = renderer.create(
      <Provider store={store}>
        <History />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
