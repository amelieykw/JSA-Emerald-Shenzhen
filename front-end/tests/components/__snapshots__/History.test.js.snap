// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<History /> <History /> should render correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 30,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 20,
      }
    }
  >
    History
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "amount": -900,
          "date": "2019-11-22",
          "remaining": 3400,
          "text": "$900 payment received",
          "type": "payment",
        },
        Object {
          "amount": 125,
          "date": "2019-01-03",
          "remaining": 4300,
          "text": "2018 yearly interest $125",
          "type": "interest",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginVertical": 5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 23,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 20,
                  "width": 45,
                },
                Object {
                  "backgroundColor": "#72BB53",
                },
              ]
            }
          >
            <Text />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              $900 payment received
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "grey",
                    "fontSize": 15,
                  },
                ]
              }
              uppercase={false}
            >
              2019-11-22
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginVertical": 5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 23,
                  "height": 45,
                  "justifyContent": "center",
                  "marginRight": 20,
                  "width": 45,
                },
                Object {
                  "backgroundColor": "#FC7043",
                },
              ]
            }
          >
            <Text />
          </View>
          <View>
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              2018 yearly interest $125
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "grey",
                    "fontSize": 15,
                  },
                ]
              }
              uppercase={false}
            >
              2019-01-03
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
